t/plugin/response-rewrite2.t

```diff
@@ -0,0 +1,516 @@
+
+repeat_each(1);
+no_long_string();
+no_shuffle();
+no_root_location();
+run_tests;
+
+__DATA__
+
+=== TEST 1:  add plugin with valid filters
+--- config
+    location /t {
+        content_by_lua_block {
+            local plugin = require("apisix.plugins.response-rewrite")
+            local ok, err = plugin.check_schema({
+                filters = {
+                    {
+                        regex = "Hello",
+                        scope = "global",
+                        replace = "World",
+                        options = "jo"
+                    }
+                }
+            })
+            if not ok then
+                ngx.say(err)
+            end
+
+            ngx.say("done")
+        }
+    }
+--- request
+GET /t
+--- response_body
+done
+--- no_error_log
+[error]
+
+
+
+=== TEST 2:  add plugin with invalid filter scope
+--- config
+    location /t {
+        content_by_lua_block {
+            local plugin = require("apisix.plugins.response-rewrite")
+            local ok, err = plugin.check_schema({
+                filters = {
+                    {
+                        regex = "Hello",
+                        scope = "two",
+                        replace = "World",
+                        options = "jo"
+                    }
+                }
+            })
+            if not ok then
+                ngx.say(err)
+            else
+                ngx.say("done")
+            end
+        }
+    }
+--- request
+GET /t
+--- response_body
+property "filters" validation failed: failed to validate item 1: property "scope" validation failed: matches none of the enum values
+--- no_error_log
+[error]
+
+
+
+=== TEST 3:  add plugin with invalid filter empty value
+--- config
+    location /t {
+        content_by_lua_block {
+            local plugin = require("apisix.plugins.response-rewrite")
+            local ok, err = plugin.check_schema({
+                filters = {
+                    {
+                        regex = "",
+                        replace = "world"
+                    }
+                }
+            })
+            if not ok then
+                ngx.say(err)
+            else
+                ngx.say("done")
+            end
+        }
+    }
+--- request
+GET /t
+--- response_body
+invalid value as filter field regex
+--- no_error_log
+[error]
+
+
+
+=== TEST 4:  add plugin with invalid filter regex options
+--- config
+    location /t {
+        content_by_lua_block {
+            local plugin = require("apisix.plugins.response-rewrite")
+            local ok, err = plugin.check_schema({
+                filters = {
+                    {
+                        regex = "hello",
+                        options = "h"
+                    }
+                }
+            })
+            if not ok then
+                ngx.say(err)
+            else
+                ngx.say("done")
+            end
+        }
+    }
+--- request
+GET /t
+--- error_code eval
```

Comment:
Why it's output is unlike the previous test?

Reply:
Yes, it's unlike. Because when invalid option, it'll abort with exception like [here](https://gist.github.com/kwanhur/dc17e10efe1a029b65225a8a9f2a58e8) and return `500` status with body default `5xx.html`.

So add prefix on `err` has no effect.

Reply:
How about use `xpcall` to catch the err?

Reply:
Yep, had updated with `pcall` and test case.